domain/tutoringModel/NestorFolder/nestor_utils.py

```diff
@@ -0,0 +1,158 @@
+                ]
+
+    return input_dataframe
+
+
+# Helper functions
+
+
+def build_bn(edges_list: int, extend_structure=False):
+    """This function reads the edges in form of a List of tuples
+    and returns a built Bayesian Network"""
+
+    bn = BayesianNetwork()
+    bn.add_edges_from(ebunch=edges_list)
+    print("\nType of Network:", type(bn))
+    print("\nSuccess! - Built Bayesian Network with ICERI\n")
+    return bn
+
+
+# def build_bn_multiple_structures(edges_list, data, topology="algorithm"):
+#     """This function reads the edges in form of a List of tuples
+#     and returns a built Bayesian Network"""
+
+#     if str(topology) == "iceri":
+#         bn = BayesianNetwork()
+#         bn.add_edges_from(ebunch=edges_list)
+#         print("\nType of Network:", type(bn))
+#         print("\nSuccess! - Built Bayesian Network with ICERI\n")
+#         return bn
+
+#     elif topology == "algorithm":
+#         est = TreeSearch(data=data)
+#         bn = est.estimate(estimator_type="chow-liu")
+#         print("\nType of Network:", type(bn))
+#         print("Success: Topology of BN built using Chow-Liu Tree Search")
+#         model = BayesianNetwork(bn.edges())
+
+#         return model
+
+#     elif topology == "hybrid":
+#         # initiate a Bayesian Network
+#         bn = BayesianNetwork()
+#         # Building Topology with Constraint Based Method
+#         print("\nTopology Part 1: Constraint Based")
+#         mmhc = MmhcEstimator(data=data)
+#         skeleton = mmhc.mmpc()
+#         # Orient the edges with a Score Based Method
+#         # Using Hill climb method
+#         hc = HillClimbSearch(data=data)
+#         bn = hc.estimate(
+#             tabu_length=10,
+#             white_list=skeleton.to_directed().edges(),
+#             scoring_method=BDeuScore(data=data),
+#             show_progress=False,
+#         )
+#         print("\nTopology Part 2: Score Based")
+#         print("\nType of Network:", type(bn))
+#         print("\nSuccess: Topology of BN built using a Hybrid Strategy")
+#         return bn
+
+#     else:
+#         print("Error: False Topology for Bayesian Network")
+
+
+# function to extend the network on top of learner data
+
+
+def extend_bn_to_le(
+    network,
+    edges_active_reflective,
+    edges_sensory_intuitive,
+    edges_visual_verbal,
+    edges_sequential_global,
+):
+    """
+    This function intends to extend the structure to LE
+    on top of the relationships between Learner Characteristics
+    """
+    # print("\nInput Type of Network:", type(network))
+    print("\nExtending the Nestor topology to Learning Elements")
+    network.add_edges_from(ebunch=edges_active_reflective)
+    network.add_edges_from(ebunch=edges_sensory_intuitive)
+    network.add_edges_from(ebunch=edges_visual_verbal)
+    network.add_edges_from(ebunch=edges_sequential_global)
+    # print("\nOutput Type of Network:", type(network))
+
+    return network
```

Comment:
i.e. one add edges call
```suggestion
    all_edges = (
        edges_active_reflective
        + edges_sensory_intuitive
        + edges_visual_verbal
        + edges_sequential_global
    )
    network.add_edges_from(all_edges)
    return network
```